include/oneapi/tbb/partitioner.h

```diff
@@ -642,18 +649,63 @@ class static_partitioner {
 class affinity_partitioner : affinity_partitioner_base {
 public:
     affinity_partitioner() {}
+    // new implementation just extends existing interface
+    typedef affinity_partition_type task_partition_type;
 
 private:
     template<typename Range, typename Body, typename Partitioner> friend struct start_for;
     template<typename Range, typename Body, typename Partitioner> friend struct start_reduce;
     template<typename Range, typename Body, typename Partitioner> friend struct start_deterministic_reduce;
     template<typename Range, typename Body, typename Partitioner> friend struct start_scan;
-    // new implementation just extends existing interface
-    typedef affinity_partition_type task_partition_type;
+
     // TODO: consider to make split_type public
     typedef affinity_partition_type::split_type split_type;
 };
 
+template<typename BasePartitioner>
+class numa_partitioner {
+    std::size_t num_numa_nodes;
+    BasePartitioner& base_partitioner;
+
+public:
+  numa_partitioner() : num_numa_nodes(get_number_of_numa_nodes()), base_partitioner() {}
+  numa_partitioner(BasePartitioner& bp) : num_numa_nodes(get_number_of_numa_nodes()), base_partitioner(bp) {}
+  
+  void initialize_arena() const {
+    for (std::size_t node = 0; node < num_numa_nodes; ++node) {
+      this->arenas.emplace_back(tbb::task_arena::constraints().set_numa_id(node));
```

Comment:
If the same instance is used across multiple parallel_fors, won't the arenas vector keep growing? I think initialize would be repeatedly invoked.